src/cargo/core/resolver/dep_cache.rs

```diff
@@ -542,9 +546,10 @@ impl RequirementError {
                     match parent {
                         None => ActivateError::Fatal(anyhow::format_err!(
                             "Package `{}` does not have feature `{}`. It has a required dependency \
-                             with that name, but only optional dependencies can be used as features.",
+                             with that name, but only optional dependencies can be used as features.{}",
                             summary.package_id(),
-                            feat
+                            feat,
+                            closest
                         )),
```

Comment:
This error message is already telling the user that something exists that could be mistaken for a feature but wasn't.  It is unlikely that a typo would happen here and it feels like the two messages don't compose well.

Reply:
We could search through all the features which exist, and find those which enable this dependency?

Fwiw, this error message would have helped with the reason I opened this PR, but that is a very niche case.

Reply:
Yes, I did. This is entirely right, and I will remove the suggestion from this branch.

(I had pattern-matched this to the thing which sometimes happens on GitHub of comments double-sending)